Source/Core/Core/GeckoCodeConfig.cpp

```diff
@@ -2,17 +2,157 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#include "Core/GeckoCodeConfig.h"
+
 #include <algorithm>
 #include <sstream>
 #include <string>
 #include <vector>
 
+#include <curl/curl.h>
+
 #include "Common/IniFile.h"
+#include "Common/Logging/Log.h"
 #include "Common/StringUtil.h"
-#include "Core/GeckoCodeConfig.h"
 
 namespace Gecko
 {
+static size_t DownloadCodesWriteCallback(void* contents, size_t size, size_t nmemb,
+                                         std::string* body)
+{
+  size_t realsize = size * nmemb;
+  body->insert(body->end(), reinterpret_cast<char*>(contents),
+               reinterpret_cast<char*>(contents) + realsize);
+  return realsize;
+}
+
+std::vector<GeckoCode> DownloadCodes(std::string gameid, bool* succeeded)
```

Comment:
This could preferably be split up into three functions.

One being game ID handling:

```cpp
static std::string RelevantName(const std::string& game_id)
{
  switch (game_id[0])
  {
  case 'R':
  case 'S':
  case 'G':
    return game_id;
  default:
    // All channels (WiiWare, VirtualConsole, etc) are identified by their first four characters
    return game_id.substr(0, 4);
}
```

One that handles the retrieval via CURL and returns the received string, and one that parses/iterates/enumerates the received string (kind of odd to do something other than downloading in a function called DownloadCodes).